src/test/scala/org/broadinstitute/dsde/workbench/sam/api/AdminUserRoutesSpec.scala

```diff
@@ -325,27 +306,78 @@ class AdminUserRoutesSpec extends AnyFlatSpec with Matchers with ScalatestRouteT
 
   "DELETE /admin/v1/user/{userSubjectId}/petServiceAccount/{project}" should "delete a pet (as an admin)" in {
     // Arrange
-    val enabledUser = Generator.genWorkbenchUserBoth.sample.get
     val samRoutes = new MockSamRoutesBuilder(allUsersGroup)
-      .withAdminUser(adminUser) // enabled "admin" user who is making the http request
-      .withEnabledUser(enabledUser) // "persisted/enabled" user we will check the status of
+      .withAdminUser() // enabled "admin" user who is making the http request
+      .withEnabledUser(defaultUser) // "persisted/enabled" user we will check the status of
       .build
     // Act
-    Delete(s"/api/admin/v1/user/${enabledUser.id}/petServiceAccount/myproject") ~> samRoutes.route ~> check {
+    Delete(s"/api/admin/v1/user/$defaultUserId/petServiceAccount/myproject") ~> samRoutes.route ~> check {
       status shouldEqual StatusCodes.NoContent
     }
   }
 
   it should "not allow a non-admin to delete a pet" in {
     // Arrange
-    val enabledUser = Generator.genWorkbenchUserBoth.sample.get
     val samRoutes = new MockSamRoutesBuilder(allUsersGroup)
-      .withNonAdminUser(enabledUser)
-      .withEnabledUser(enabledUser) // "persisted/enabled" user we will check the status of
+      .withNonAdminUser()
+      .withEnabledUser(defaultUser) // "persisted/enabled" user we will check the status of
       .build
     // Act
     Delete(s"/api/admin/v1/user/$defaultUserId/petServiceAccount/myproject") ~> samRoutes.route ~> check {
       status shouldEqual StatusCodes.Forbidden
     }
   }
+
+  "GET /admin/v2/user/{userId}" should "get the corresponding user object (as an admin)" in {
+    // Arrange
+    val samRoutes = new MockSamRoutesBuilder(allUsersGroup)
+      .withAdminUser() // enabled "admin" user who is making the http request
+      .withEnabledUser(defaultUser) // "persisted/enabled" user we will check the status of
+      .build
+
+    // Act and Assert
+    Get(s"/api/admin/v2/user/$defaultUserId") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.OK
+      responseAs[SamUser] shouldEqual defaultUser
+    }
+  }
+
+  it should "not allow access to a user object (as an non admin)" in {
+    // Arrange
+    val samRoutes = new MockSamRoutesBuilder(allUsersGroup)
+      .withEnabledUser(defaultUser) // "persisted/enabled" user we will check the status of
+      .withNonAdminUser()
+      .build
+
+    // Act and Assert
+    Get(s"/api/admin/v2/user/$defaultUserId") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.Forbidden
+    }
+  }
+
+  it should "not allow access even if a user object does not exist (as an admin)" in {
```

Comment:
Instead of me micromanaging your test names, do you mind taking one last look through and making sure the naming of tests and test setup is consistent? I think it's fine as it is but could be even better :) For instance little things like "return a user object" or something instead of just "allow access," and trying to make the admin/non admin distinction easy to parse at a glance. Overall I think these tests look good though!